docs/ref/contrib/gis/gdal.txt

```diff
@@ -1832,3 +1832,25 @@ Settings
 A string specifying the location of the GDAL library.  Typically,
 this setting is only used if the GDAL library is in a non-standard
 location (e.g., ``/home/john/lib/libgdal.so``).
+
+Exceptions
+==========
+
+.. exception:: GDALException
+
+    The base GDAL exception, indicates a GDAL-related error.
+
+    .. note::
+
+        ``OGRException`` is a legacy name for this exception only kept for
+        backwards compatibility.
+
+.. exception:: OGRIndexError
+
+    An exception raised when an invalid result is returned from the underlying
+    library.
```

Comment:
This exception seems like it's not just thrown as a result of failures in the library, but could be a client error. For example. if you instantiate a `DataSource` but give an invalid index to `__getitem__` it will throw an `OGRIndexError`. In fact, it seems to be used pretty much the same way I'd expect to see `IndexError` used.